### PR TITLE
[FLINK-32860][flink-table-api] Add missing visibility annotation for Table APIs

### PR DESCRIPTION
**Pull request by LadyForest**
## What is the purpose of the change

This pull request adds missing visibility annotations for classes under `flink-table-api-java`, `flink-table-api-java-bridge`, and `flink-table-common`.  Meanwhile, it also introduces a new architectural rule to perform detection.

## Brief change log

This PR contains 4 commits
  - 8c51e837 adds missing annotations for classes under `flink-table-api-java-bridge`
  - 433dabb3 adds missing annotations for classes under `flink-table-common`
  - 571f1b7e adds missing annotations for classes under `flink-table-api-java`
  - 1d99ae10 introduces a new rule `ALL_CLASSES_IN_TABLE_API_SHOULD_HAVE_VISIBILITY_ANNOTATIONS`

## Verifying this change

This change added tests and can be verified by running `ArchitectureTest`. Some violations are fixed because since this PR. 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no) some classes are marked with `@PublicEvolving`
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by lincoln-lil — COMMENTED** [review]
@LadyForest thanks for contributing this! Overall looks good, I left some comment.

File: flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/lookup/cache/trigger/PeriodicCacheReloadTrigger.java [hunk 1]
```diff
@@ -128,6 +131,7 @@ public static PeriodicCacheReloadTrigger fromConfig(ReadableConfig config) {
      * ScheduledExecutorService#scheduleWithFixedDelay(Runnable, long, long, TimeUnit) and {@link
      * ScheduledExecutorService#scheduleAtFixedRate(Runnable, long, long, TimeUnit)}}.
      */
+    @Internal
```
[the remark this answers is not in the document]
Comment (LadyForest) [reply flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/lookup/cache/trigger/PeriodicCacheReloadTrigger.java:134]: good catch!

File: flink-architecture-tests/flink-architecture-tests-production/src/main/java/org/apache/flink/architecture/rules/TableApiRules.java [hunk 2]
```diff
@@ -39,6 +48,10 @@
 
 /** Rules for Table API. */
 public class TableApiRules {
+
+    private static final Pattern TABLE_API_MODULES =
+            Pattern.compile(".*/flink-table-(api-java|api-java-bridge|common)/.*");
```
[the remark this answers is not in the document]
Comment (LadyForest) [reply flink-architecture-tests/flink-architecture-tests-production/src/main/java/org/apache/flink/architecture/rules/TableApiRules.java:53]: I think so. Nit: it should be `flink-table-api-bridge-base`

**Comment by LadyForest** [comment]
- Rebase `origin/master,` in case newly introduced APIs are missing annotation.
- Fix visibility annotation for `PeriodicCacheReloadTrigger$ScheduleMode`.
- Include `flink-table-api-bridge-base` to the rule pattern.

**Review by lincoln-lil — APPROVED** [review]
LGTM

**Pull request merged**